filebeat/beater/filebeat.go

```diff
@@ -51,32 +51,33 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 
 	// count active events for waiting on shutdown
 	wgEvents := &sync.WaitGroup{}
-	finishedLogger := newFinishedLogger(wgEvents)
+	registrarLogger := newLogger(wgEvents)
 
 	// Setup registrar to persist state
-	registrar, err := registrar.New(config.RegistryFile, finishedLogger)
+	registrar, err := registrar.New(config.RegistryFile, registrarLogger)
 	if err != nil {
 		logp.Err("Could not init registrar: %v", err)
 		return err
 	}
 
 	// Make sure all events that were published in
-	registrarChannel := newRegistrarLogger(registrar)
+	publisherOutput := newPublisherOutput(registrar.GetInput())
 
-	// Channel from spooler to harvester
-	publisherChan := newPublisherChannel()
+	// Channel from spooler to publisher
+	spoolerOutput := newSpoolerOutput()
 
 	// Publishes event to output
-	publisher := publisher.New(config.PublishAsync, publisherChan.ch, registrarChannel, b.Publisher)
+	publisher := publisher.New(config.PublishAsync, spoolerOutput.ch, publisherOutput, b.Publisher)
 
 	// Init and Start spooler: Harvesters dump events into the spooler.
-	spooler, err := spooler.New(config, publisherChan)
+	spooler, err := spooler.New(config, spoolerOutput)
```

Comment:
idea of using `publisherChan` (while naming is bad), was to make more clear which component the spooler outputs to.